src/api.ts

```diff
@@ -325,6 +325,63 @@ type CreateAttachmentParams = {
   metadata?: Maybe<Record<string, any>>;
 };
 
+export class SharedCache {
+  private static instance: SharedCache | null = null;
```

Comment:
Let's simplify even more:
- have a `sharedCache.ts` file
- and apply the singleton pattern https://www.patterns.dev/vanilla/singleton-pattern/